### PR TITLE
TAS: Make ResourceFlavorSpec that defines `topologyName` immutable

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
TAS: Make ResourceFlavorSpec that defines `topologyName` immutable

#### Which issue(s) this PR fixes:

Part of #3733 

#### Does this PR introduce a user-facing change?

```release-note
ResourceFlavorSpec that defines topologyName is not immutable
```

### CODE REVIEW

File: test/integration/tas/tas_test.go [hunk 1]
```diff
@@ -80,6 +80,58 @@ var _ = ginkgo.Describe("Topology Aware Scheduling", ginkgo.Ordered, func() {
 			tasFlavor := testing.MakeResourceFlavor("tas-flavor").TopologyName("default").Obj()
 			gomega.Expect(k8sClient.Create(ctx, tasFlavor)).Should(gomega.HaveOccurred())
 		})
+
+		ginkgo.It("should not allow to update ResourceFlavorSpec", func() {
```
Comment (mbobrovskyi) [inline test/integration/tas/tas_test.go:84]: I would propose to create separate describe for validation. Like we have for topology on L1040.

File: test/integration/tas/tas_test.go [hunk 2]
```diff
@@ -80,6 +80,58 @@ var _ = ginkgo.Describe("Topology Aware Scheduling", ginkgo.Ordered, func() {
...
+
+			ginkgo.By("Deleting one of the nodeLabels", func() {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(tasFlavor), tasFlavor)).To(gomega.Succeed())
+				tasFlavor.Spec.NodeLabels = map[string]string{
+					"foo": "bar",
+				}
+				gomega.Expect(k8sClient.Update(ctx, tasFlavor)).Should(gomega.HaveOccurred())
+			})
+
+			ginkgo.By("Cleaning up RF", func() {
```
Comment (mbobrovskyi) [inline test/integration/tas/tas_test.go:131]: I think it's better to use defer

```
			err := k8sClient.Create(ctx, tasFlavor)
			if err == nil {
				defer func() {
					util.ExpectObjectToBeDeleted(ctx, k8sClient, tasFlavor, true)
				}()
			}
```

File: test/integration/tas/tas_test.go (hunk 1 above)
Comment (PBundyra) [reply test/integration/tas/tas_test.go]: Done

File: test/integration/tas/tas_test.go (hunk 2 above)
Comment (PBundyra) [reply test/integration/tas/tas_test.go]: Done

**Comment by mimowo** [comment]
/approve
leaving lgtm to @mbobrovskyi

**Review by mbobrovskyi — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**